src/libos/crates/host-socket/src/stream/states/connect.rs

```diff
@@ -87,13 +94,15 @@ impl<A: Addr + 'static, R: Runtime> ConnectingStream<A, R> {
             req.io_handle.take();
 
             if retval == 0 {
+                arc_self.connected.store(true, Ordering::Relaxed);
```

Comment:
Add `io_handle.is_some() `conditional judgment to prevent repeated submit connect request.